service/controller/app/resource/chart/desired.go

```diff
@@ -111,6 +110,123 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 	return chartCR, nil
 }
 
+func (r *Resource) pickRepositoryURL(ctx context.Context, cc *controllercontext.Context, cr v1alpha1.App, chartName string) (string, error) {
+	if len(cc.Catalog.Spec.Repositories) == 1 {
+		return cc.Catalog.Spec.Repositories[0].URL, nil
+	}
+
+	var chart v1alpha1.Chart
+	err := cc.Clients.K8s.CtrlClient().Get(
+		ctx,
+		types.NamespacedName{Name: chartName, Namespace: r.chartNamespace},
+		&chart,
+	)
+	if apierrors.IsNotFound(err) || tenant.IsAPINotAvailable(err) {
+		// Repositories is guaranteed by Custom Resource Definition to have at least one entry.
```

Comment:
Should we really rely on this? I think we should fall back to `repository` if `repositories[0]` is nil. Otherwise we will need to transition all app CRs somehow before we roll this out.

Reply:
The `repositories` array is in Catalog CRD. We will need to transition all Catalog CRs (copy stuff in `storage` to `repositories[0]`), but I think that's an ok thing to do. WDYT?
I can add a fallback to `storage`, but that's incurring technical debt - we would have to update the code again after we deprecate `storage`.

Reply:
If we can control all catalog CRs, then that is fine to me. Some customers are creating catalogs now, so it might be some effort to try and get them to move their source of truth over. Maybe there is a metric or something that let's us check how many catalogs actually exist?

Reply:
Let me check, but there should be a reasonable number of clusters for us to "migrate". I can update `kubectl-gs` to generate the new Catalog CR format.

Reply:
I checked and there are currently 7 custom Catalog CRs (excluding default Giant Swarm ones):
```
===> bumblebee
NAMESPACE    NAME                                  CATALOG URL                                                                 AGE
default      nfs-subdir-external-provisioner       https://kubernetes-sigs.github.io/nfs-subdir-external-provisioner/          118d
--
===> fox
NAMESPACE    NAME                                  CATALOG URL                                                                 AGE
default      hephy                                 https://charts.teamhephy.com/                                               88d
--
===> gauss
NAMESPACE    NAME                                  CATALOG URL                                                                 AGE
default      demo-catalog-lj                       https://ljakimczuk.github.io/test-helm-repo/                                34d
default      machine-learning                      https://pipo02mix.github.io/ml-catalog/                                     206d
--
===> ginger
NAMESPACE    NAME                                  CATALOG URL                                                                 AGE
org-ross     hephy                                 https://charts.teamhephy.com/                                               125d
--
===> helios
NAMESPACE    NAME                                  CATALOG URL                                                                 AGE
default      thg-example-app-catalog               https://thehutgroup.github.io/example-app-catalog                           20d
--
===> tiger
NAMESPACE    NAME                                  CATALOG URL                                                                 AGE
default      hdi-apps-sandbox                      https://pg2000.github.io/tkp-app-catalog/                                   68d
```

Reply:
In the end I added a `switch/case` which takes Catalogs without `.spec.repositories` but `.spec.storage` defined into account. Better be safe than sorry.